### PR TITLE
Weather app - Marjaana Tienaho

### PR DESCRIPTION
**Pull request by MayaBlixt**
Link to my deployed project:
https://fabweatherapp.netlify.app/
My reflections on how this week's project turned out:

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <!--Google fonts-->
+    <link href="https://fonts.googleapis.com/css?family=Roboto+Slab|Ruda" rel="stylesheet">
+    <!--Favicon sun-->
+    <link rel="apple-touch-icon" sizes="180x180" href="./assets/apple-touch-icon.png">
+    <link rel="icon" type="image/png" sizes="32x32" href="./assets/favicon-32x32.png">
+    <link rel="icon" type="image/png" sizes="16x16" href="./assets/favicon-16x16.png">
```
Comment (BamStugendal) [inline index.html:12]: You have two favicon here, which one are you using? Maybe you don't need both of them?

File: style.css [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+.weather-img {
+    width: 100%;
+    display: flex;
+    justify-content: space-around;
+    align-items: flex-end;
+}
+
+#temperature {
+    font-size: 70px;
+}
```
Comment (BamStugendal) [inline style.css:32]: You could add a class to the "temperature" so that you use class in css and id in javascript. 😄

File: style.css [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    display: none;
+}
+
+#short-footer {
+    display: block;
+    text-align: center;
+}
+
+/* MEDIA QUERIES tablet */
+@media (min-width: 667px) and (max-width: 1024px) {
```
Comment (BamStugendal) [inline style.css:97]: Nice and smooth transition between mobile, tablet, and desktop view. Good job! 😄

File: style.css [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+
+    }
+
+    .weather-img,
+    .city-container,
+    .sun-set-container {
+        display: flex;
+        flex-direction: row;
+        align-items: center;
+    }
```
Comment (BamStugendal) [inline style.css:175]: Nice to see that you gather the selector that will have the same property in one 👍

File: weather.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        const sunsetValue = new Date(json.sys.sunset * 1000)
+        const sunsetTime = sunsetValue.toLocaleTimeString([], {
+            hour: '2-digit',
+            minute: '2-digit'
+        })
+        sunrise.innerText = `Sunrise: ${sunriseTime}`
+        sunset.innerText = `Sunset: ${sunsetTime}`
+    })
+    .catch((err) => {
+        console.log('Caught error: ${err}')
```
Comment (BamStugendal) [inline weather.js:45]: Consider using `console.error` instead of `console.log`

File: weather.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+const API_URL_WEATHER = `https://api.openweathermap.org/data/2.5/weather?q=Stockholm,Sweden&units=metric&APPID=${API_KEY}`
+const API_URL_FORECAST = `https://api.openweathermap.org/data/2.5/forecast?q=Stockholm,Sweden&units=metric&APPID=${API_KEY}`
+
+const temperature = document.getElementById('temperature')
+const mainIcon = document.getElementById('city-icon')
+const cityName = document.getElementById('city-name')
+const description = document.getElementById('description')
+const sunrise = document.getElementById('sunrise')
+const sunset = document.getElementById('sunset')
+const forecastContent = document.getElementById('main-forecast-data')
```
Comment (BamStugendal) [inline weather.js:11]: Nice that you gather the variables here! Cleaner code 👍  😄

File: weather.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+            hour: '2-digit',
+            minute: '2-digit'
+        })
+        const sunsetValue = new Date(json.sys.sunset * 1000)
+        const sunsetTime = sunsetValue.toLocaleTimeString([], {
+            hour: '2-digit',
+            minute: '2-digit'
+        })
+        sunrise.innerText = `Sunrise: ${sunriseTime}`
+        sunset.innerText = `Sunset: ${sunsetTime}`
```
Comment (BamStugendal) [inline weather.js:42]: You could take this out from .then and make a function so that the .then looks cleaner 😄

File: weather.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+    .then((json) => {
+        //Present the temperature (rounded to 1 decimal place), the city name and what type of weather it is
+        temperature.innerHTML = `${json.main.temp.toFixed(1)}°C`
+        cityName.innerHTML = `${json.name}`
+        description.innerHTML = `${json.weather[0].description}`
+
+        //Set the current weather image
+        const weatherImgConst = json.weather[0].icon;
+        mainIcon.src =`./assets/${weatherImgConst}.png`
```
Comment (BamStugendal) [inline weather.js:28]: Love the weather icon! so cute 😍

File: index.html (hunk 1 above)
Comment (BamStugendal) [reply index.html:12]: nice icon though! 😄

**Review by BamStugendal — APPROVED** [review]
Great job Marjaana! Love the cute weather icons and the smooth transition between the different views!

File: style.css (hunk 2 above)
Comment (MayaBlixt) [reply style.css:32]: Updated to use class in css and id in javascript :)

File: index.html (hunk 1 above)
Comment (MayaBlixt) [reply index.html:12]: Removed favicon 16x16 :)

File: style.css (hunk 3 above)
Comment (MayaBlixt) [reply style.css:97]: Thank you :)

File: style.css (hunk 4 above)
Comment (MayaBlixt) [reply style.css:175]: Thanks 😄

File: weather.js (hunk 5 above)
Comment (MayaBlixt) [reply weather.js:45]: Yes, that is true. Typo ;)

Comment (MayaBlixt) [reply weather.js:45]: Updated

File: weather.js (hunk 6 above)
Comment (MayaBlixt) [reply weather.js:11]: 😄

File: weather.js (hunk 8 above)
Comment (MayaBlixt) [reply weather.js:28]: Thank you 😍

**Comment by MayaBlixt** [comment]
Updated as commented above

File: index.html [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    <title>Weather app</title>
+</head>
+
+<body>
+    <div class="container">
+        <section class="main-city-container">
+            <div class="weather-container">
+                <div class="weather-img">
+                    <p class="temperature" id="temperatureId">Loading temperature...</p>
+                    <img class="city-icon" id="cityIconId" src="" alt="Current weather" />
```
Comment (EmOlsz) [inline index.html:22]: No need for empty `src` attribute here - you can create it in JS

File: style.css [hunk 10]
```diff
@@ -0,0 +1,235 @@
+html,
+body {
+    height: 100%;
+    position: relative;
+    box-sizing: border-box;
+}
+
+body {
+    background-image: url(./assets/Background.jpg);
```
Comment (EmOlsz) [inline style.css:9]: It's more safe to wrap url in quotes, some browsers might have problem with reading path without them

```
background-image: url("./assets/Background.jpg");
```

Moreover, capital letters shouldn't be used for names of files if they are not React components

File: style.css [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+    background-position: cover;
+    margin: 0;
+    Font-Family: 'Roboto Slab', Serif;
```
Comment (EmOlsz) [inline style.css:12]: it's `font-family`, not `Font-family` and `serif`, not `Serif`

File: weather.js [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+        const sunsetValue = new Date(json.sys.sunset * 1000)
+        const sunsetTime = sunsetValue.toLocaleTimeString([], {
+            hour: '2-digit',
+            minute: '2-digit'
+        })
+        sunrise.innerText = `Sunrise: ${sunriseTime}`
+        sunset.innerText = `Sunset: ${sunsetTime}`
+    })
+    .catch((err) => {
+        console.error('Caught error: ${err}')
```
Comment (EmOlsz) [inline weather.js:44]: Great :)

**Review by EmOlsz — APPROVED** [review]
Nice project Marjaana! I like your code structure, it's well done. Moreover, CSS aspects are also nice (some issues there, but you will find details in comments).
Speaking of them, please take a look on each occurrence and consider them as some tips for next projects :)

**Pull request closed without merging**